src/main/java/frc/robot/Autonomous.java

```diff
@@ -0,0 +1,168 @@
+package frc.robot;
+
+import com.revrobotics.RelativeEncoder;
+// import com.revrobotics.SparkMaxRelativeEncoder;
+
+// SUBSYSTEM IMPORTS
+import frc.robot.subsystems.BallPath.BallPath.BallAction;
+import frc.robot.subsystems.BallPath.Shooter.Shooter.ShotPosition;
+import frc.robot.subsystems.BallPath.BallPathImpl;
+import frc.robot.subsystems.Drivetrain.DriveImpl;
+
+import com.revrobotics.SparkMaxPIDController;
+// import com.revrobotics.CANSparkMax.ControlType;
+
+
+import edu.wpi.first.math.controller.PIDController;
+
+// import java.util.concurrent.TimeUnit;
+// import ca.team3161.lib.robot.TitanBot;
+
+public class Autonomous {
+
+    private DriveImpl drivetrain;
+    // private Shooter shooter;
+    // private Intake intake;
+    // private Elevator elevator;
+    private BallPathImpl ballPath;
+    private double wheelCircumference = Math.pow((Math.PI*2), 2);
+    private double distance;
+    private double setPoint;
+    private double max = 1000;
+    private double min = 0;
+
+    private double kP = 0.00001;
+    private double kI = 0;
+    private double kD = 0;
+
+    private SparkMaxPIDController leftPIDController;
+    private SparkMaxPIDController rightPIDController;
+
+    private RelativeEncoder leftEncoder;
+    private RelativeEncoder rightEncoder;
+
+    // private double maxSpeed = 1;
+    private double zRotation = 0;
+    private double targetDistance;
+
+    private PIDController positionPIDController;
+
+
+
+
+
+    public Autonomous(DriveImpl drivetrain, BallPathImpl ballPath){
+        this.drivetrain = drivetrain;
+        this.ballPath = ballPath;
+        this.distance = 0;
+        this.setPoint = 0;
+
+        this.leftEncoder = drivetrain.getEncoder(1);
+        this.rightEncoder = drivetrain.getEncoder(0);
+        this.positionPIDController = new PIDController(kP, kI, kD);
+        // this.leftPIDController = drivetrain.getController(1).getPIDController();
+        // this.rightPIDController = drivetrain.getController(0).getPIDController();
+
+        leftPIDController.setP(kP);
+        leftPIDController.setI(kI);
+        leftPIDController.setD(kD);
+
+        rightPIDController.setP(kP);
+        rightPIDController.setI(kI);
+        rightPIDController.setD(kD);
+
+
+        leftPIDController.setOutputRange(min, max);
+        rightPIDController.setOutputRange(min, max);
```

Comment:
All of this `leftPIDController`/`rightPIDController` stuff should be removed. Right now this is going to `NullPointerException` as soon as it's initialized because `leftPIDController` is uninitialized/null and then you try to `setP()` on it. And, as we discussed in person. we don't want to modifying the PID constants of the velocity loops anyway.

Reply:
oh lol I meant to do that and just forgot one sec